### ISSUE TITLE
different outputs for keras.layers.TimeDistributed()

### ISSUE THREAD

**Issue by aabramovrepo** [matterport/Mask_RCNN#2762]
moving from 

`Ubuntu 18.04 | CUDA 10.1 | tensorflow 2.1.0 | keras 2.3`

   to

`Ubuntu 20.04 | CUDA 11.1 | tensorflow 2.4.0 | keras 2.4.0`

I'm getting a different output in

    x = KL.TimeDistributed(KL.Conv2D(fc_layers_size, (pool_size, pool_size), padding='valid'), name='mrcnn_class_conv1')(x)

namely, in the first setup:

    x = Tensor("mrcnn_class_conv1/Reshape_1:0", shape=(None, 1000, 1, 1, 1024), dtype=float32)

and in the second setup:

    x = KerasTensor(type_spec=TensorSpec(shape=(1, None, 1, 1, 1024), dtype=tf.float32, name=None), name='conv2d/squeeze_batch_dims/Reshape_1:0', description="created by layer 'conv2d'")

tensor shapes are different somehow: `(1, None, 1, 1, 1024)` instead of `(None, 1000, 1, 1, 1024)`.

Does anybody have an idea about reasons and possible fixes? Does this migration make sense?

Thanks in advance for any advice!
